src/dotnet/APIView/apiview.yml

```diff
@@ -30,230 +30,373 @@ pr:
+                    steps:
+                    - checkout: self
+
+                    - download: current
+                      artifact: APIView
+                      timeoutInMinutes: 5
+
+                    - task: FileTransform@1
+                      displayName: 'Update Angular Config for Environment'
+                      inputs:
+                        folderPath: '$(Pipeline.Workspace)/APIView/**/APIViewWeb.zip'
+                        fileType: json
+                        targetFiles: wwwroot/spa/assets/config.json
+
+                    - task: AzureRmWebAppDeployment@4
+                      displayName: 'Deploy Azure App Service'
+                      inputs:
+                        azureSubscription: 'APIview staging resource group'
+                        appType: 'webApp'
+                        WebAppName: 'apiviewstaging'
+                        packageForLinux: '$(Pipeline.Workspace)/APIView/**/APIViewWeb.zip'
+                        ScriptType: 'Inline Script'
+                        InlineScript: |
+                          @echo off
+                          echo Installing npm packages
+                          call npm install --prefix D:\home\site\wwwroot --force
+                          call npm run-script build --prefix D:\home\site\wwwroot --force
+                          call D:\home\site\wwwroot\Python\python -m pip uninstall api-stub-generator
+                          call D:\home\site\wwwroot\Python\python -m pip install pylint==2.13.9 pylint-guidelines-checker==0.0.6 api-stub-generator==0.3.2 --index-url "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+                        enableCustomDeployment: true
+                        DeploymentType: zipDeploy
+      
+      - ${{ if and(in(variables['Build.Reason'], 'Manual', ''), eq(variables['System.TeamProject'], 'internal')) }}:
+        - stage: Test_UI_Publish
+          displayName: Publish Test UI
+          dependsOn: Build
+
+          jobs:
+            - deployment: Publish_Test_UI
+              displayName: Publish Test UI
+              pool:
+                name: azsdk-pool-mms-ubuntu-2004-general
+                image: azsdk-pool-mms-ubuntu-2004-1espt
+                os: linux
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                    - checkout: self
+
+                    - download: current
+                      artifact: APIView
+                      timeoutInMinutes: 5
+
+                    - task: AzureRmWebAppDeployment@4
+                      displayName: 'Azure App Service Deploy: apiviewuat'
+                      inputs:
+                        azureSubscription: 'APIViewUI-Instance-deployment-connection'
+                        WebAppName: apiviewuat
+                        packageForLinux: $(Pipeline.Workspace)/APIView/**/*.zip
+
+        - stage: Production_Publish
```

Comment:
Can we add a condition to skip any runs of this pipeline stage from PR? This is just to avoid prod deployment if anyone is trying to deploy changes to staging or test from a PR.

Reply:
It falls under the `      - ${{ if and(in(variables['Build.Reason'], 'Manual', ''), eq(variables['System.TeamProject'], 'internal')) }}:`

Reply:
You can see this in the `APIView - CI` run.